src/emulator/functionsEmulator.ts

```diff
@@ -821,24 +853,24 @@ export class FunctionsEmulator implements EmulatorInstance {
    *  specified in extension.yaml. This will ALWAYS be populated when emulating extensions, even if they
    *  are using the default version.
    */
-  askInstallNodeVersion(cwd: string, nodeMajorVersion?: number): string {
-    const pkg = require(path.join(cwd, "package.json"));
+  getNodeBinary(backend: EmulatableBackend): string {
+    const pkg = require(path.join(backend.functionsDir, "package.json"));
     // If the developer hasn't specified a Node to use, inform them that it's an option and use default
-    if ((!pkg.engines || !pkg.engines.node) && !nodeMajorVersion) {
+    if ((!pkg.engines || !pkg.engines.node) && !backend.nodeMajorVersion) {
       this.logger.log(
         "WARN",
-        "Your functions directory does not specify a Node version.\n   " +
+        `Your functions directory ${backend.functionsDir} does not specify a Node version.\n   ` +
           "- Learn more at https://firebase.google.com/docs/functions/manage-functions#set_runtime_options"
       );
       return process.execPath;
     }
 
     const hostMajorVersion = process.versions.node.split(".")[0];
-    const requestedMajorVersion: string = nodeMajorVersion
-      ? `${nodeMajorVersion}`
+    const requestedMajorVersion: string = backend.nodeMajorVersion
+      ? `${backend.nodeMajorVersion}`
       : pkg.engines.node;
     let localMajorVersion = "0";
-    const localNodePath = path.join(cwd, "node_modules/.bin/node");
+    const localNodePath = path.join(backend.functionsDir, "node_modules/.bin/node");
```

Comment:
Honestly, Im not sure either - I left this because I wanted to avoid breaking any corner cases and I assumed it was there for a good reason.  Maybe @abeisgoat knows?